docker/Dockerfile

```diff
@@ -0,0 +1,30 @@
+
+FROM alpine:latest AS builder
+
+ARG BUILD_TYPE
+ARG RELEASE
+
+RUN wget https://github.com/qbittorrent/qBittorrent/archive/refs/tags/release-${RELEASE}.tar.gz && \
+    tar xf release-${RELEASE}.tar.gz && \
+    cd qBittorrent-release-${RELEASE} && \
+    apk add --no-cache qt6-qttools-dev g++ libtorrent-rasterbar-dev cmake boost-dev ninja && \
+    cmake -B build-nox -G "Ninja" -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DGUI=OFF -DQT6=ON -DSTACKTRACE=OFF && \
```

Comment:
What about creating a native alpine package and install it? The main question is would there be any downsides of it?
I think it would be better than to list all lib requirements and specific build commands in this dockerfile.

If it is doable I'll look into it myself.

Reply:
I think it's more trouble than it's worth, I've never created a package with alpine, but it will require more or less the same content of what's already in the dockerfile (so one more file to maintain with the same content), plus everything that's needed to create the package itself, furthermore well'need to list all the requirement (for both compilation and installation) in the APKBUILD (alpine's file to give to whatever is building their packages) rather than list in the build container the requirement to build qBittorrent and in the last docker image the requirement to run the container.
Plus it will make the docker image heavier by having the same files both in the package and installed (essentially the same file twice)

Reply:
alpine already has an apkbuild/package for it, for whoever wants to use it (so there is no need to carry this here in the qbittorrent repo as well, just this dockerfile):  

https://pkgs.alpinelinux.org/package/edge/community/x86_64/qbittorrent  
https://git.alpinelinux.org/aports/tree/community/qbittorrent/APKBUILD  

currently, this involves using a `FROM alpine:edge`, then `apk add qbittorrent-nox` should work  

(speaking as the maintainer of this in alpine)

Reply:
>alpine already has an apkbuild/package for it, for whoever wants to use it (so there is no need to carry an apkbuild here in the qbittorrent repo as well, just this dockerfile):

Note that for qbt, it would be best if we can have complete control over compile flags and library versions but those are non-targets for this first-step PR. I plain to look into it after this PR is merged.

>once alpine-3.16 is released, it will also be part of stable.

Good to know!